### PR TITLE
Fixed Xcode 14 compilation: avoid `@available` properties

### LINKED ISSUE #1660: [Xcode 14] Stored properties cannot be marked potentially unavailable with '@available'
- [x] I have updated Purchases SDK to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/RevenueCat/purchases-ios/blob/main/Development/CONTRIBUTING.md)
- [x] I have searched the [Community](https://community.revenuecat.com)
- [x] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [x] I have searched for [existing Github issues](https://github.com/RevenueCat/purchases-ios/issues)

**Describe the bug**
Refuses to build with Xcode 14b1 due to requirement for stored properties to not have availability markings

1. Environment
   1. Platform: Xcode 14 (new beta)
   2. SDK version: 4.5.1
   3. StoreKit 2 enabled (Y/N): n/a
   4. OS version: n/a
   5. Xcode version: Xcode 14 beta 1
   6. How widespread is the issue. Percentage of devices affected: 100% unable to build
2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue
3. Steps to reproduce, with a description of expected vs. actual behavior
4. **Other information** (e.g. stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc.)

**Additional context**

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes #1660 and [SDKONCALL-24].
This is the same approach used in #1596.

[SDKONCALL-24]: https://revenuecats.atlassian.net/browse/SDKONCALL-24?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Sources/Support/BeginRefundRequestHelper.swift [hunk 1]
```diff
@@ -24,17 +24,30 @@ class BeginRefundRequestHelper {
...
 #if os(iOS)
+
+    private let _sk2Helper: Any!
+
     @available(iOS 15.0, *)
     @available(macOS, unavailable)
     @available(watchOS, unavailable)
     @available(tvOS, unavailable)
-    lazy var sk2Helper = SK2BeginRefundRequestHelper()
+    var sk2Helper: SK2BeginRefundRequestHelper {
```
Comment (taquitos) [inline Sources/Support/BeginRefundRequestHelper.swift:34]: image

Danger Zone

**Comment by NachoSoto** [comment]
The `else` is broken, fixing it.

**Comment by NachoSoto** [comment]
Okay this is ready now.

**Review by taquitos — APPROVED** [review]
🚢 🐐

**Comment by aboedo** [comment]
awesome! should we prepare a release for it?

**Comment by taquitos** [comment]
I think since this makes it compile in Xcode 14, we should do a release, it could be blocking customers from testing 😄

**Comment by taquitos** [comment]
Yeah, I'm just thinking about how we can cut down any further noise to support.

**Comment by NachoSoto** [comment]
That's fair. Okay I can do a `4.5.2`

**Comment by NachoSoto** [comment]
But let me see if I can wrap up #1666, it would be nice to include too.

**Pull request merged**